COMP371/ShadowMap.cpp

```diff
@@ -6,44 +6,49 @@
 
 #include "RenderingContext.h"
 #include "ChunkManager.h"
+#include "ShaderProgram.h"
 
 #include <iostream>
 
 ShadowMap::ShadowMap(uint32 width, uint32 height, glm::vec3 lightDirection)
 	:m_width(width)
 	,m_height(height)
 {
+	m_shadowShader = RenderingContext::get()->shaderCache.loadShaderProgram("sm_shader", "shadowmap_vert.glsl", "shadowmap_frag.glsl");
 	updateMvp(lightDirection);
-	buildBuffer();
+	buildFBO();
 }
 
-void ShadowMap::updateMvp(glm::vec3 lightDirection)
+void ShadowMap::updateMvp(const glm::vec3& lightDirection)
 {
-	glm::vec3 invLightDirection = glm::normalize(-lightDirection);
+	glm::vec3 invLightDirection = glm::normalize(-lightDirection) * 300.0f;
 	glm::vec3 playerPos = RenderingContext::get()->camera.transform.getPosition();
-	glm::vec3 lightPosition = (invLightDirection*300.0f);
+	glm::vec3 lightPosition = { playerPos.x + invLightDirection.x, invLightDirection.y, playerPos.z + invLightDirection.z };
 	float32 loadingRadius = (ChunkManager::LOADINGRADIUS + 0.5) * ChunkManager::CHUNKWIDTH;
 
-	glm::mat4 proj = glm::ortho<float>(-loadingRadius, loadingRadius, 0, ChunkManager::CHUNKHEIGHT, 0.1f , 500.0f);
-	glm::mat4 view = glm::lookAt(lightPosition, glm::vec3(0, 0, 0), glm::vec3(0, 1, 0));
+	glm::mat4 proj = glm::ortho<float>(-(loadingRadius+50), (loadingRadius+50), 0, ChunkManager::CHUNKHEIGHT, 0.1f , 500.0f);
```

Comment:
A REAL programmer would have added a comment explaining that the +50 is to expand the viewing box from the point of view of the sun (so that the shadows don't end flush with the chunks).